x/subscription/keeper/subscription.go

```diff
@@ -107,6 +116,13 @@ func (k Keeper) CreateSubscription(
 		return utils.LavaFormatWarning("create subscription failed", err)
 	}
 
+	// subscription looks good; let's create it and charge the creator
+	price := plan.GetPrice()
+	price.Amount = price.Amount.MulRaw(int64(duration))
```

Comment:
to keep the denom:
price.Amount = price.Amount.MulRaw(int64(duration)) ->
price= price.MulRaw(int64(duration))